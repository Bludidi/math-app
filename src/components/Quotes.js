import React, { useState, useEffect } from 'react';
import axios from 'axios';
import './quotes.css';

const Quotes = () => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const apiUrl = 'https://api.api-ninjas.com/v1/quotes';
    const apiKey = 'O6cCUithwsgOIfcYuEV0Vw==3YIjiRhzwxmoi13s';
    const params = {
      category: 'knowledge',
      limit: 1,
    };

    const headers = {
      'X-Api-Key': `${apiKey}`,
    };

    axios
      .get(apiUrl, { params, headers })
      .then((response) => {
        setData(response.data);
        setLoading(false);
      })
      .catch((error) => {
        error('Error fetching data:', error);
      });
  }, []);

  const handleRefresh = () => {
    window.location.reload();
  };

  return (
    <div className="quotes">
      <h1>QUOTE</h1>
      <br />
      {loading ? (
        <p>Loading...</p>
      ) : (
        <>
          {data.map((item) => (
            <div key={item.id} className="quote_div">
              <p>{item.quote}</p>
              <span>
                -
                {item.author}
              </span>
            </div>
          ))}
          <button type="button" onClick={handleRefresh} className="q-btn">
            Next Quote
          </button>
        </>
      )}
    </div>
  );
};

export default Quotes;
