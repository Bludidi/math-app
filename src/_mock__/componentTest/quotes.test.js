import { render } from '@testing-library/react';
import Quotes from '../../components/quotes';

describe('Quote', () => {
  test('should render quotes', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });
});
