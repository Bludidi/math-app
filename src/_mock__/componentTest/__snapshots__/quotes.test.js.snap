// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render quotes 1`] = `
<div>
  <div
    class="quotes"
  >
    <q>
      Mathematic is not about numbers, equations, computations, or algorithms: it is about understanding.
    </q>
    <br />
    <h4>
      -William Paul Thurston
    </h4>
  </div>
</div>
`;
